amt.go

```diff
@@ -200,14 +203,65 @@ func (n *Node) get(bs Blocks, height int, i uint64, out interface{}) error {
 		}
 	}
 
-	subn, err := n.loadNode(bs, subi)
+	subn, err := n.loadNode(bs, subi, false)
 	if err != nil {
 		return err
 	}
 
 	return subn.get(bs, height-1, i%nodesForHeight(width, height), out)
 }
 
+func (r *Root) Delete(i uint64) error {
+	if err := r.Node.delete(r.bs, int(r.Height), i); err != nil {
+		return err
+	}
+	r.Count--
+
+	for r.Node.Bmap[0] == 1 {
```

Comment:
Something about this `1` feels at least counter-intuitive. Is it that if only 1 bit is set then we unpack the level below? Shouldn't that be doing `bits.OnesCount == 1` then? (and `bits.TrailingZeros` to get the index later)

Reply:
(https://golang.org/pkg/math/bits/, also, both of those calls should get compiled down to single x86 instructions (`popcnt`, `tzcnt`))

Reply:
This is specifically if the lowest bit is set only.

Looking at this though, I think that it doesn't account for the lowest level having only a single item in that slot.